disable LOCALECOMPARE on macOS with CMake (#2957)
mixxxdj/mixxx: CMakeLists.txt

# Locale Aware Compare for SQLite
find_package(SQLite3)
# FIXME
# It is difficult to get qmake to link Qt to a custom built SQLite on macOS instead of the system SQLite,
# which results in a crash on startup when LOCALECOMPARE is enabled.

> if (APPLE)

Reviewer: Please use function-style if for this macro, i.e. no space.

> # FIXME
> # It is difficult to get qmake to link Qt to a custom built SQLite on macOS instead of the system SQLite,
> # which results in a crash on startup when LOCALECOMPARE is enabled.
> if(APPLE)
>   cmake_dependent_option(LOCALECOMPARE "Locale Aware Compare support for SQLite" OFF "SQLite3_FOUND" OFF)
> else()
>   cmake_dependent_option(LOCALECOMPARE "Locale Aware Compare support for SQLite" ON "SQLite3_FOUND" OFF)
> endif()

Reviewer 2: ```suggestion
# FIXME: It is difficult to get qmake to link Qt to a custom built SQLite on
# macOS instead of the system SQLite, which results in a crash on startup when
# LOCALECOMPARE is enabled.
cmake_dependent_option(LOCALECOMPARE "Locale Aware Compare support for SQLite" ON "SQLite3_FOUND; NOT APPLE" OFF)
```

See https://cmake.org/cmake/help/latest/module/CMakeDependentOption.html for details

Author: I thought about doing this, but if someone were to try to fix this, they'd have to revert that first instead of using `-DLOCALECOMPARE=ON`.

Reviewer: I don't have no clear preference. Please discuss and then merge.

Reviewer 2: Are you sure that it's not possible to override the default value with `-D`. IRC it works.

Author: I'm pretty sure:
> This macro presents an option to the user only if a set of other conditions are true.

Reviewer 2: Yes, that means it won't show up as an option in cmake-gui, but I think you can still override it anyway.

Author: No, I just tested `cmake_dependent_option(LOCALECOMPARE "Locale Aware Compare support for SQLite" ON "SQLite3_FOUND; NOT UNIX" OFF)` and manually setting `-DLOCALECOMPARE=ON` and it did not turn on LOCALECOMPARE. So I think we should merge this as it is.
